src/test/regress/expected/join_pushdown.out

```diff
@@ -152,7 +152,7 @@ ORDER BY 1;
 -- Full outer join with different distribution column types, should error out
 SELECT * FROM test_table_1 full join test_table_2 using(id);
 ERROR:  cannot push down this subquery
-DETAIL:  Shards of relations in subquery need to have 1-to-1 shard partitioning
+DETAIL:  test_table_1 and test_table_2 are not colocated
```

Comment:
maybe run https://github.com/citusdata/citus/tree/main/src/test/regress/citus_tests/query_generator on null-shard-key tables

Reply:
(@aykut-bozkurt is doing some testing based on query generator, fyi)